src/arrayMethodReduce.js

```diff
@@ -1,11 +1,15 @@
 'use strict';
 
-/**
- * Implement method Reduce
- */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let accumulator = initialValue !== undefined ? initialValue : this[0];
+    const start = initialValue !== undefined ? 0 : 1;
```

Comment:
In this case, if the user passes `undefined` as an `initialValue` it won't work correctly.
You could check if the `initialValue` was passed in this way: `if (arguments.length < 2) { ... }`.